tests/test_plugins/test_mode_solver.py

```diff
@@ -812,7 +812,7 @@ def test_mode_solver_nan_pol_fraction():
     md = ms.solve()
     check_ms_reduction(ms)
 
-    assert list(np.where(np.isnan(md.pol_fraction.te))[1]) == [8, 9]
+    assert list(np.where(np.isnan(md.pol_fraction.te))[1]) == [9]
```

Comment:
@dbochkov-flexcompute I see you added this test. I changed the formula in `eigs_to_effective_index`, and as a result sometimes evanescent modes which were previously `nan` are no longer `nan`. As a result I had to modify this test. Could you take a look and see if this makes sense?

Reply:
yeah, it seems fine. This test just checks that if there are modes with nan's in pol_fraction, they are put at the end of list